### PR TITLE
Support test names by comment

### LINKED ISSUE #141: Support naming conformance test assertions
Conformance tests should be able to have a name, such that the combination of file name, test name, and specific assertion (like "cannot-convert:Object? to Object!") is unique.

That allows tests to be better documented, and allows us to stop relying on the line number to disambiguate in reports, which is brittle to unrelated changes to the test file.

It is important that discovering the name not require parsing of the Java syntax, but be entirely textual.

The test framework would assert that the combination of test name and assertion be unique within each file for assertions associated with a test name.

Any assertion not associated with a test name would report the line number of the line being tested, as it does today.

### PR DESCRIPTION
**Pull request by netdpb**
Partially fixes #141.

### CODE REVIEW

**Comment by netdpb** [comment]
Ping for review.

File: conformance-test-framework/src/main/java/org/jspecify/conformance/ExpectedFact.java [hunk 1]
```diff
@@ -108,35 +124,91 @@ static class Reader {
...
+      return checkUniqueTestNames(expectedFacts.build());
+    }
+
+    private void setTestName(@Nullable String testName) {
+      if (testName == null) {
+        return;
+      }
+      check(this.testName == null, "test name already set");
+      check(facts.isEmpty(), "test name must come before assertions for a line");
+      this.testName = testName.trim();
```
Comment (wmdietl) [inline conformance-test-framework/src/main/java/org/jspecify/conformance/ExpectedFact.java]: Should this first check that the test name isn't empty?

**Review by wmdietl — APPROVED** [review]
One possible corner case to look at, otherwise looks good.

File: conformance-test-framework/src/main/java/org/jspecify/conformance/ExpectedFact.java (hunk 1 above)
Comment (netdpb) [reply conformance-test-framework/src/main/java/org/jspecify/conformance/ExpectedFact.java]: Good catch!
